enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -87,17 +90,25 @@ type Provider struct {
 	streamableImages []string
 }
 
-func NewProvider(env environment.Env, imageCacheAuthenticator *container.ImageCacheAuthenticator, buildRoot string) (*Provider, error) {
-	if len(*streamableImages) > 0 {
+func runSociStore(ctx context.Context) {
+	for {
 		log.Infof("Starting soci store")
-		sociStoreDir := "/var/lib/soci-store/store"
-		cmd := exec.CommandContext(env.GetServerContext(), "soci-store", sociStoreDir)
+		sociStoreDir := sociStorePath
+		cmd := exec.CommandContext(ctx, "soci-store", sociStoreDir)
 		logWriter := log.Writer("[socistore] ")
 		cmd.Stderr = logWriter
 		cmd.Stdout = logWriter
-		if err := cmd.Start(); err != nil {
-			return nil, status.UnavailableErrorf("could not start soci store: %s", err)
-		}
+		cmd.Run()
+
+		log.Infof("Detected soci store crash, restating")
+		// If the store crashed, the path must be unmounted to recover.
+		syscall.Unmount(sociStorePath, 0)
```

Comment:
We should add some kind of cooldown here (even if it's a simple `time.Sleep(500 * time.Millisecond)` for now) in case socistore is crash looping.

Reply:
Yeah, sounds good